MERLin/MERLin/Module.swift

```diff
@@ -8,32 +8,51 @@
 
 import RxSwift
 
+public struct PageInfo: Equatable {
+    public let pageName: String
+    public let section: String
+    public let pageType: String
+    
+    init(withPage page: PageRepresenting) {
+        pageName = page.pageName
+        section = page.section
+        pageType = page.pageType
+    }
+}
+
 public enum ViewControllerEvent: EventProtocol, Equatable {
     case uninitialized
     case initialized
     case appeared
     case disappeared
     
-    case newViewController(UIViewController & PageRepresenting, events: Observable<ViewControllerEvent>)
+    case newViewController(PageInfo, events: Observable<ViewControllerEvent>)
     
     public static func == (lhs: ViewControllerEvent, rhs: ViewControllerEvent) -> Bool {
         switch (lhs, rhs) {
         case (.uninitialized, .uninitialized),
              (.initialized, .initialized),
              (.appeared, .appeared),
              (.disappeared, .disappeared): return true
-        case let (.newViewController(lhsController, _), .newViewController(rhsController, _)): return lhsController == rhsController
+        case let (.newViewController(lhsPageInfo, _), .newViewController(rhsPageInfo, _)):
+            return lhsPageInfo == rhsPageInfo
         default: return false
         }
     }
 }
 
-public protocol PageRepresenting: class {
+public protocol PageRepresenting {
     var pageName: String { get }
     var section: String { get }
     var pageType: String { get }
 }
 
+public extension PageRepresenting {
+    func pageInfo() -> PageInfo {
```

Comment:
i would make it computed var
`var pageInfo: PageInfo`

Reply:
nope. It might return different values wether the properties are computed or not.